packages/gestalt/src/IconButton.js

```diff
@@ -87,9 +107,75 @@ const IconButtonWithForwardRef: React$AbstractComponent<
   const classes = classnames(styles.button, touchableStyles.tapTransition, {
     [styles.disabled]: disabled,
     [styles.enabled]: !disabled,
-    [touchableStyles.tapCompress]: !disabled && isTapping,
+    [touchableStyles.tapCompress]:
+      props.role !== 'link' && !disabled && isTapping,
   });
 
+  function handleClick(event) {
+    if (onClick) {
+      onClick({ event });
+    }
+  }
+
+  function handleLinkClick({ event }) {
+    handleClick(event);
+  }
+
+  const renderPogComponent = (selected?: boolean): Node => {
+    return (
+      <Pog
+        active={!disabled && isActive}
+        bgColor={bgColor}
+        dangerouslySetSvgPath={dangerouslySetSvgPath}
+        focused={!disabled && isFocusVisible && isFocused}
+        hovered={!disabled && isHovered}
+        icon={icon}
+        iconColor={iconColor}
+        padding={padding}
+        selected={selected}
+        size={size}
+      />
+    );
+  };
+
+  if (props.role === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        accessibilityLabel={accessibilityLabel}
+        disabled={disabled}
+        inline
```

Comment:
@AlbertCarreras could you double check whether this is what we want? Currently on the `IconButton role="button"` we set it to `display="block"`

![image](https://user-images.githubusercontent.com/127199/92253518-74bf2880-ee84-11ea-91c1-5f62907025c3.png)

But we set it to `inline` for `role="link"`

![image](https://user-images.githubusercontent.com/127199/92253587-8acce900-ee84-11ea-9916-584169f96fe0.png)

Example:
```
function Example() {

  return (
    <>
      <IconButton accessibilityLabel='Default IconButton' icon='share' onClick={() => {}} />
      <IconButton accessibilityLabel='Default IconButton' icon='share' onClick={() => {}} />
      <IconButton accessibilityLabel='Link IconButton' icon='link' role="link" target="blank" href="https://www.pinterest.com" />
      <IconButton accessibilityLabel='Link IconButton' icon='link' role="link" target="blank" href="https://www.pinterest.com" />
    </>
  );
}
```

Results in this:
![image](https://user-images.githubusercontent.com/127199/92253707-b4861000-ee84-11ea-995f-37aaa77d264d.png)

Reply:
DONE